common/include/pcl/experimental/executor/cuda_executor.hpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#ifdef __CUDACC__
+  #include <cuda_runtime_api.h>
+#endif
+
+#include <pcl/experimental/executor/property.h>
+#include <pcl/experimental/executor/type_trait.h>
+
+namespace executor {
+
+#ifdef __CUDACC__
+template <typename F>
+__global__ void global_kernel(F f) {
+  f();
+}
+#endif
+
+template <typename Blocking, typename ProtoAllocator>
+struct cuda_executor;
+
+#ifdef __CUDACC__
+template <>
+struct is_executor_available<cuda_executor> : std::true_type {};
+#endif
+
+template <typename Blocking = blocking_t::always_t,
+          typename ProtoAllocator = std::allocator<void>>
+struct cuda_executor {
+  struct shape_type {
+    struct dim3 {
+      unsigned int x, y, z;
+    } grid_size, block_size;
+  };
+
+  template <typename Executor, InstanceOf<Executor, cuda_executor> = 0>
+  friend bool operator==(const cuda_executor&,
```

Comment:
constexpr?

Also, the return checks if the incoming executor is also blocking with a default allocator. If so, why not just have a simple `friend bool operator==(const cuda_executor&, const cuda_executor&) { return true; }`?

Reply:
This would not allow checks between two different executor types such as `inline_executor` and `cuda_executor`.